src/fr/animesama/src/eu/kanade/tachiyomi/extension/fr/animesama/AnimeSama.kt

```diff
@@ -73,84 +75,175 @@ class AnimeSama : ParsedHttpSource() {
             .sorted()
             .map { chapter ->
-                SChapter.create().apply {
-                    name = "Chapitre $chapter"
-                    setUrlWithoutDomain(javascriptUrl + "/$chapter")
+            }.asReversed()
+        var chapList: MutableList<SChapter> = ArrayList()
+        var retard = 0
+
+        Log.v("AnimeSama", document.html())
+        if (document.html().containsMultipleTimes("resetListe()")) {
+            var list = document.html().split("resetListe();").toMutableList()
+            list.removeAt(0)
+            list.removeAt(0)
+            list[0] = list[0].split("finirList")[0]
+            list = list[0].split(");").toMutableList()
+            Log.v("AnimeSama", list.toString())
+            list.forEach { command ->
+                when {
+                    command.contains("creerListe(") -> {
+                        var clean = command.replace("creerListe(", "").trimIndent().split(",")
+                        var start = clean[0].replace(" ", "").toInt()
+                        var end = clean[1].replace(" ", "").toInt()
+
+                        for (i in start..end) {
+                            Log.v("AnimeSama", "" + (chapList.size + 1) + " - " + i)
+                            chapList.add(
+                                SChapter.create().apply {
+                                    name = "Chapitre " + (i)
+                                    setUrlWithoutDomain(document.baseUri() + "/" + javascriptUrl + "&id=${chapList.size + 1}")
+                                },
+                            )
+                        }
+                    }
+                    command.contains("newSP(") -> {
+                        var clean = command.replace("newSP(", "").trimIndent()
+                        Log.v("AnimeSama", "" + (chapList.size + 1) + " - " + clean)
+                        chapList.add(
+                            SChapter.create().apply {
+                                name = "Chapitre " + (clean)
+                                setUrlWithoutDomain(document.baseUri() + "/" + javascriptUrl + "&id=${chapList.size + 1}")
+                            },
+                        )
+                        retard++
+                    }
+                    /* The site contains an as-yet unused command called "newSPE", and as I have no concrete examples of its use, I haven't implemented it yet. */
                 }
             }
+        }
+        for (index in chapList.size..toJson.size - 1) {
+            Log.v("AnimeSama", "" + (chapList.size + 1) + " - " + (chapList.size + 1 - retard))
+            chapList.add(
+                SChapter.create().apply {
+                    name = "Chapitre " + (chapList.size + 1 - retard)
+                    setUrlWithoutDomain(document.baseUri() + "/" + javascriptUrl + "&id=${chapList.size + 1}")
+                },
+            )
+        }
```

Comment:
this can probably use regex to extract specific things, instead of splitting and indexing?